### PR TITLE
Feature/lp 37 refactor account controller

### PR DESCRIPTION
**Pull request by harveymw27**
Account controller refactored for vertical-slice pattern

### CODE REVIEW

**Review by YogirajA — CHANGES_REQUESTED** [review]
@harveymw27  Registration issue:
If anyone tries to register with a username that already exists, the API correctly doesn't register the user correctly but frontend flow somehow logs them into an existing account anyway. This must be fixed asap. It may have existed before this. This must be fixed on the frontend as well as on the API.

File: src/API/LeadershipProfileAPI/appsettings.json [hunk 1]
```diff
@@ -11,5 +11,8 @@
     "Sender": "\"Leadership Portal\" <noreply@test.com>",
     "Username": "",
     "Password": ""
+  },
+  "ApplicationConfiguration": {
+    "WebApplicationUrl": "https://localhost:5001"
```
[the remark this answers is not in the document]
Comment (harveymw27) [reply src/API/LeadershipProfileAPI/appsettings.json:16]: done

File: src/API/LeadershipProfileAPI/Features/Account/AccountController.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            if (response.Result)
+            {
+                return Ok(response);
+            }
+
+            return BadRequest(response);
+        }
+    }
+
+    public class ForgotPasswordModel
```
[the remark this answers is not in the document]
Comment (harveymw27) [reply src/API/LeadershipProfileAPI/Features/Account/AccountController.cs:131]: I will do them all then.

File: src/API/LeadershipProfileAPI/Features/Account/Delete.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+using System.Threading.Tasks;
+using LeadershipProfileAPI.Data;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+
+namespace LeadershipProfileAPI.Features.Account
+{
+    public class Delete
```
[the remark this answers is not in the document]
Comment (harveymw27) [reply src/API/LeadershipProfileAPI/Features/Account/Delete.cs:13]: I was using it for testing. I left it there should users need to be removed. Do you want it removed?

File: src/API/LeadershiProfileAPI.Tests/Features/AccountTests.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
+﻿using LeadershipProfileAPI.Controllers;
```
[the remark this answers is not in the document]
Comment (harveymw27) [reply src/API/LeadershiProfileAPI.Tests/Features/AccountTests.cs:1]: done

File: src/API/LeadershipProfileAPI/Features/Account/Delete.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (harveymw27) [reply src/API/LeadershipProfileAPI/Features/Account/Delete.cs]: Removed from code base

**Comment by harveymw27** [comment]
> @harveymw27 Registration issue:
> If anyone tries to register with a username that already exists, the API correctly doesn't register the user correctly but frontend flow somehow logs them into an existing account anyway. This must be fixed asap. It may have existed before this. This must be fixed on the frontend as well as on the API.

I am creating a defect ticket for this as it did exist

**Review by YogirajA — APPROVED** [review]

**Pull request merged**